bundles/org.openhab.binding.modbus.lambda/src/main/java/org/openhab/binding/modbus/lambda/internal/handler/BoilerHandler.java

```diff
@@ -0,0 +1,585 @@
+                }
+                if (poller != null) {
+                    logger.trace("Boiler: Es wird gepollt }");
+                    poller.poll();
+                }
+            }
+        } else {
+            // logger.trace("Boiler: handleCommand: Es wird geschrieben, GroupID: {}, command {}",
+            // channelUID.getGroupId(), command);
+            try {
+
+                if (GROUP_BOILER_REG50.equals(channelUID.getGroupId())) {
+                    // logger.trace("Boiler: im BOILER_REG50 channelUID {} ", channelUID.getIdWithoutGroup());
+
+                    switch (channelUID.getIdWithoutGroup()) {
+
+                        case CHANNEL_BOILER_MAXIMUM_BOILER_TEMPERATURE:
+
+                            // logger.trace("Boiler: command: {}", command);
+                            writeInt16(reg50baseadress, getScaledInt16Value(command));
+                            break;
+
+                    }
+                }
+
+            }
+
+            catch (LambdaException error) {
+                if (hasConfigurationError() || getThing().getStatus() == ThingStatus.OFFLINE) {
+                    return;
+                }
+                String cls = error.getClass().getName();
+                String msg = error.getMessage();
+
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        String.format("Error with: %s: %s", cls, msg));
+            }
+        }
+    }
+
+    /**
+     * Initialization: Load the config object of the block Connect to the slave
+     * bridge Start the periodic polling
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(BoilerConfiguration.class);
+        // logger.debug("Initializing thing with properties: {}", thing.getProperties());
+
+        startUp();
+    }
+
+    /**
+     * Adresses for the polling registers, used for reading and writing
+     */
+    private int baseadress;
+    private int reg50baseadress;
+
+    /**
+     * This method starts the operation of this handler Connect to the slave bridge
+     * Start the periodic polling1
+     */
+    private void startUp() {
+        if (comms != null) {
+            return;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bridge is offline");
+            return;
+        }
+
+        try {
+            slaveId = slaveEndpointThingHandler.getSlaveId();
+
+            comms = slaveEndpointThingHandler.getCommunicationInterface();
+        } catch (EndpointNotInitializedException e) {
+            // this will be handled below as endpoint remains null
+        }
+
+        if (comms == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
```

Comment:
If you are sure this can't be null. Don;t supress, but use `Objects.requireNonNull()`

Reply:
C&P from the stiebeleltron-bundle. 
I am not capable of changing it.

Reply:
It tells the compiler to ignore that label can be null. If this suppression was not there, the compiler would complain.
Now that it is surpressed, it is harder to fix whenever it becomes null. When using `Objects.requireNonNull()` ytou have more control and can add a custom message whenver it hits null. Making it easier to track and fix in the future.

Reply:
Do you think:
```
   if (comms == null) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
            logger.debug("CommunicationInterface is null, Thing & Bridge are offline");
            return;
        }
```
is OK? I found this code in 
`openhab-addons\bundles\org.openhab.binding.modbus.kermi\src\main\java\org\openhab\binding\modbus\kermi\internal\handler\KermiXcenterThingHandler.java`

Reply:
If this code is OK with you:
I still have to add it to the other handlers an delete the commentary.
Please give feedback!